### ISSUE TITLE
Graceful Handling of Hanging Backups

### ISSUE THREAD

**Issue by lnhsingh** [cockroachdb/docs#7930]
When Backups hang silently, even if there is a legit reason that will be cleaned up soon, customers get nervous because it means a potential loss of data. Consistency in backups being taken gives them confidence that their data is safe.

We want to have some way for backups to succeed in light of hanging--overriding priority, etc. so that data is secure.

Fixes a known issue where backup performance can appear to be almost stuck in tables that have lots of intents.

Backup is almost unusably slow in some circumstances, and users do not have a good way of knowing what is causing this performance degradation. This is an attempt to address one known cause of backup performance degradation.

See this in registration, but also other tables as well.

Note: @pbardea added the current workaround to the [on-call runbook](https://cockroachlabs.atlassian.net/wiki/spaces/TS/pages/269156516/Bulk-IO+and+Schema+Change+Runbook).

cc: @mwang1026

**Comment by mwang1026** [comment]
cc @dt  who's working on this functionality with KV

**Comment by jseldess** [comment]
@lnhsingh, @mwang1026, ahead of docs being ready, I'm trying to add this to the 20.2 release notes summary, but I don't really know what to say. I don't see any linked eng PRs here or in the jira epic. What exactly is changed that users should know about?

**Comment by mwang1026** [comment]
https://github.com/cockroachdb/cockroach/pull/51624

tl;dr backups used to hang forever / until the transaction holding open the intent closed, which would cause a lot of headaches and confusion. Now, we queue up that data to try backing up later, and then if X time has passed and the intent still hasn't closed then we run the backup export requests at HIGH PRIORITY which will cause the timestamp of the transaction holding the intent to be kicked (and a transaction retry)
